src/engine/renderer/tr_shader.cpp

```diff
@@ -5992,209 +5990,154 @@ A second parameter will cause it to print in sorted order
+	stageTypeStrings[stageType_t::ST_PHYSICALMAP] = "PHYSICALMAP";
+	stageTypeStrings[stageType_t::ST_SPECULARMAP] = "SPECULARMAP";
+	stageTypeStrings[stageType_t::ST_REFLECTIONMAP] = "REFLECTIONMAP";
+	stageTypeStrings[stageType_t::ST_REFRACTIONMAP] = "REFRACTIONMAP";
+	stageTypeStrings[stageType_t::ST_DISPERSIONMAP] = "DISPERSIONMAP";
+	stageTypeStrings[stageType_t::ST_SKYBOXMAP] = "SKYBOXMAP";
+	stageTypeStrings[stageType_t::ST_SCREENMAP] = "SCREENMAP";
+	stageTypeStrings[stageType_t::ST_PORTALMAP] = "PORTALMAP";
+	stageTypeStrings[stageType_t::ST_HEATHAZEMAP] = "HEATHAZEMAP";
+	stageTypeStrings[stageType_t::ST_LIQUIDMAP] = "LIQUIDMAP";
+	stageTypeStrings[stageType_t::ST_LIGHTMAP] = "LIGHTMAP";
+	stageTypeStrings[stageType_t::ST_COLLAPSE_lighting_PBR] = "LIGHTING_PBR";
+	stageTypeStrings[stageType_t::ST_COLLAPSE_lighting_PHONG] = "LIGHTING_PHONG";
+	stageTypeStrings[stageType_t::ST_COLLAPSE_reflection_CB] = "REFLECTION_CB";
+	stageTypeStrings[stageType_t::ST_ATTENUATIONMAP_XY] = "ATTENUATIONMAP_XY";
+	stageTypeStrings[stageType_t::ST_ATTENUATIONMAP_Z] = "ATTENUATIONMAP_XZ";
+
+	std::string stageTypeHeader = "stageType";
+	size_t maxStageTypeStringLen = stageTypeHeader.length();
+
+	for ( const auto& kv : stageTypeStrings )
+	{
+		maxStageTypeStringLen = std::max( maxStageTypeStringLen, kv.second.length() );
+	}
+
+	std::string interactLightHeader = "interactLight";
+	int maxInteractLightStringLen = interactLightHeader.length();
+
+	std::string separator = " ";
+
+	std::string header = Str::Format(
+		"%-*s" "%s%-*s" "%s%-*s" "%s%-*s" "%s%s",
+		maxShaderTypeStringLen, shaderTypeHeader,
+		separator, maxShaderSortStringLen, shaderSortHeader,
+		separator, maxInteractLightStringLen, interactLightHeader,
+		separator, maxStageTypeStringLen, stageTypeHeader,
+		separator, "stageNumber:shaderName" );
+
+	std::string lineSeparator( header.length(), '-' );
+
+	Log::CommandInteractionMessage( lineSeparator );
+	Log::CommandInteractionMessage( header );
+	Log::CommandInteractionMessage( lineSeparator );
+
+	int stageCount = 0;
+	uint8_t highestShaderStageCount = 0;
+
+	for ( int i = 0; i < tr.numShaders; i++ )
+	{
+		shader_t *shader = ri.Cmd_Argc() > 2 ? tr.sortedShaders[ i ] : tr.shaders[ i ];
+
+		// Only display shaders starting with prefix if prefix is not empty.
+		if ( prefix && !Com_Filter( prefix, shader->name, false ) )
 		{
 			continue;
 		}
 
-		std::string str;
-
-		switch ( shader->type )
-		{
-			case shaderType_t::SHADER_2D:
-				str += "2D   ";
-				break;
+		highestShaderStageCount = std::max( highestShaderStageCount, shader->numStages );
 
-			case shaderType_t::SHADER_3D_DYNAMIC:
-				str += "3D_D ";
-				break;
-
-			case shaderType_t::SHADER_3D_STATIC:
-				str += "3D_S ";
-				break;
-
-			case shaderType_t::SHADER_LIGHT:
-				str += "ATTN ";
-				break;
-		}
+		std::string foundShaderTypeString = shaderTypeStrings[ shader->type ];
+		std::string foundShaderSortString = shaderSortStrings[ (shaderSort_t)shader->sort ];
```

Comment:
Note that I'm not happy with that cast. For some reason `shader->sort` is a `float`. Before the code was comparing `shader->sort` with `Util::ordinal(shaderSort_t::SS_BAD)`, now we are comparing `(shaderSort_t)shader->sort` with `Util::ordinal(shaderSort_t::SS_BAD)`, and due to nature of floats, I'm not very confident with it.

Anyway this work.

Note: I have on a branch a commit to attempt to turn `shader->sort` into a `shaderSort_t` but that's out of topic of this PR.